testcases/Lcov.py

```diff
@@ -0,0 +1,126 @@
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+'''
+Lcov setup Testcase
+--------------------
+
+This test case to setup Lcov 
+'''
+
+# FIXME: Add a smaller version of this test to the normal host test suite
+# FIXME: Work out a way to add this to the skiroot test suite.
+
+import unittest
+import os
+import time
+import OpTestConfiguration
+from common.OpTestSystem import OpSystemState
+
+import logging
+import OpTestLogger
+log = OpTestLogger.optest_logger_glob.get_logger(__name__)
+
+
+class LcovSetup(unittest.TestCase):
+    '''
+    Use the ``ppc64_cpu`` utility to turn SMT on/off and set to each possible
+    SMT mode, which is effectively a CPU hotplug operation.
+    Then start hotplugging CPU cores by using the ``ppc64_cpu`` utility to
+    turn only a specific number of cores on.
+    '''
+
+    def setUp(self):
+        conf = OpTestConfiguration.conf
+        self.cv_HOST = conf.host()
+        self.cv_IPMI = conf.ipmi()
+        self.cv_SYSTEM = conf.system()
+
+    def runTest(self):
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.c = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+        self.c.run_command("uname -a")
+        res = self.c.run_command("cat /etc/os-release | grep NAME | head -1")
+        if 'SLES' in res[0].strip():
+            self.distro = 'SLES'
+        elif 'Red Hat' in res[0].strip():
+            self.distro = 'RHEL'
+        if self.distro == 'RHEL':
+            cmd = "yum install"
+        elif self.distro == 'SLES':
+            cmd = "zypper install"
+        dep_packages = ['perl*', 'tiny*']
+        for pkg in dep_packages:
+            self.c.run_command(cmd + pkg)
+        time.sleep(5)
+        url = 'git clone https://github.com/linux-test-project/lcov.git'
+        self.c.run_command(cmd)
+        os.chdir('lcov')
+        self.c.run_command("make install")
+
+
+class LcoveCounterZero(unittest.TestCase):
+    '''
+    make the lcov counter to zero before gathering the data
+    '''
+    def setup_test(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+
+    def runTest(self):
+        '''
+        Running the test
+        '''
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.c = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+        self.c.run_command("echo 1 > /sys/kernel/debug/gcov/reset")
+        self.c.run_command("lcov --zerocounters")
+
+
+class LcovGatherData(unittest.TestCase):
+    '''
+    collect all gcov data and create an html file
+    '''
+    def setup_test(self):
+        conf = OpTestConfiguration.conf
+        self.cv_SYSTEM = conf.system()
+
+    def runTest(self):
+        '''
+        Running the test
+        '''
+        self.cv_SYSTEM.goto_state(OpSystemState.OS)
+        self.c = self.cv_SYSTEM.cv_HOST.get_ssh_connection()
+        src_path = '/root/rpmbuild/BUILD/kernel-5.14.0-283.el9/linux-5.14.0-283.el9.ppc64le'
```

Comment:
Will need a way to make this dynamic. Currently this is hardcoded

Reply:
we have handled it little differently.